### PR TITLE
threadutil: fix doc: syntax of start_thread; explain when raise_in_th…

### LINKED ISSUE #47: threadutil.raise_in_thread: need doc about when and why it does not work.

### PR DESCRIPTION
**Pull request by drmingdrmer**
syntax of start_thread.
explain when raise_in_thread does not work.

close #47

### CODE REVIEW

File: threadutil/README.md [hunk 1]
```diff
@@ -74,31 +74,45 @@ t = threadutil.start_thread(work, name='my_thread', args=(10, ), kwargs={'x': 2}
     A dictionary of keyword arguments for the target invocation. Defaults to `{}`.
 
 - `daemon`:
-    Whether to create a daemon thread.
+    Whether to create a **daemon** thread.
+
+    > A daemon thread will quit when the main thread in a process quits.
```
Comment (shuoqingding) [inline threadutil/README.md]: 应该是 when the main program quits 吧？

File: threadutil/README.md [hunk 2]
```diff
@@ -74,31 +74,45 @@ t = threadutil.start_thread(work, name='my_thread', args=(10, ), kwargs={'x': 2}
...
+    > A non-daemon thread keeps running after main thread quits.
+    > A process does not quit if there are any threads running.
```
Comment (shuoqingding) [inline threadutil/README.md]: *any non-daemon threads running?

File: threadutil/README.md [hunk 3]
```diff
@@ -74,31 +74,45 @@ t = threadutil.start_thread(work, name='my_thread', args=(10, ), kwargs={'x': 2}
...
+
 Please note that the exception will be raised only when executing python bytecode.
 If your thread calls a native/built-in blocking function, the exception will be
 raised only when execution returns to the python code.
 
-**Syntax**
+Thus This function does not guarantee that a running thread will be
+interrupted and shut down when it is called.
+If a thread has not returned from `time.sleep()` or `threading.Thread.join()`,
+it continues running until returning from these function.
```
Comment (shuoqingding) [inline threadutil/README.md]: The exception will be raised only when executing python bytecode. If your thread calls a native/built-in blocking function (such as `time.sleep`), the exception will be raised only when execution returns to the python code.
There is also an issue if the built-in function internally calls PyErr_Clear(), which would effectively cancel your pending exception. You can try to raise it again.

File: threadutil/README.md (hunk 1 above)
Comment (shuoqingding) [inline threadutil/README.md]: 或者 main process

Comment (drmingdrmer) [inline threadutil/README.md]: 就是1个process里的main thread..就1个process啊..

File: threadutil/README.md (hunk 2 above)
Comment (drmingdrmer) [inline threadutil/README.md]: 去掉keep?

File: threadutil/README.md (hunk 3 above)
Comment (drmingdrmer) [inline threadutil/README.md]: 好复杂.我直接copy了

Comment (drmingdrmer) [inline threadutil/README.md]: 酱?
```
The exception will be raised only when executing python bytecode. If your
thread calls a native/built-in blocking function (such as time.sleep), the
exception will be raised only when execution returns to the python code.

There is also an issue if the built-in function internally calls
PyErr_Clear(), which would effectively cancel your pending exception. You can
try to raise it again.

Thus This function does not guarantee that a running thread will be
interrupted and shut down when it is called.
If a thread has not returned from `time.sleep()` or `threading.Thread.join()`,
it continues running until returning from these function.

```

**Pull request merged**